### PR TITLE
Automatically generate Capitano Docs configuration

### PR DESCRIPTION
**Pull request by vipulgupta2048**
PR enables docs configuration needed for CLI documentation to be generated automatically. 

Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipulgupta2048@gmail.com>

### CODE REVIEW

File: automation/capitanodoc/capitanodoc.ts [hunk 1]
```diff
@@ -27,224 +28,106 @@ import { MarkdownFileParser } from './utils';
...
+	util: 'Utilities',
+};
+
+// Fetch all available commands
+const allCommandsPaths = new GlobSync('build/commands/**/*.js', {
+	ignore: 'build/commands/internal/**',
+}).found;
+
+// Throw error if any commands found outside of command directories
+//
```
Comment (myarmolinsky) [inline automation/capitanodoc/capitanodoc.ts:76]:
```suggestion

```

File: automation/capitanodoc/capitanodoc.ts [hunk 2]
```diff
@@ -27,224 +28,106 @@ import { MarkdownFileParser } from './utils';
...
+capitanoDoc.categories = capitanoDoc.categories.sort((a, b) =>
+	a.title.localeCompare(b.title),
+);
+
+// Sort Category file paths alhpabetically
+capitanoDoc.categories.forEach((category) => {
+	category.files.sort((a, b) => a.localeCompare(b));
+});
+
+console.log(capitanoDoc.categories);
```
Comment (myarmolinsky) [inline automation/capitanodoc/capitanodoc.ts:126]: console.log

File: automation/capitanodoc/capitanodoc.ts [hunk 3]
```diff
@@ -27,224 +28,106 @@ import { MarkdownFileParser } from './utils';
...
+
+	if (!categoriesMap[heading]) {
+		categoriesMap[heading] = { title: heading, files: [] };
+		capitanoDoc.categories.push(categoriesMap[heading]);
+	}
+
+	categoriesMap[heading].files.push(commandPath);
+}
+
+// Sort Category titles alhpabetically
```
Comment (myarmolinsky) [inline automation/capitanodoc/capitanodoc.ts:116]:
```suggestion
// Sort Category titles alphabetically
```

File: automation/capitanodoc/capitanodoc.ts [hunk 4]
```diff
@@ -27,224 +28,106 @@ import { MarkdownFileParser } from './utils';
...
+capitanoDoc.categories = capitanoDoc.categories.sort((a, b) =>
+	a.title.localeCompare(b.title),
+);
+
+// Sort Category file paths alhpabetically
```
Comment (myarmolinsky) [inline automation/capitanodoc/capitanodoc.ts:121]:
```suggestion
// Sort Category file paths alphabetically
```

File: automation/capitanodoc/capitanodoc.ts [hunk 5]
```diff
@@ -27,224 +28,106 @@ import { MarkdownFileParser } from './utils';
...
+for (const commandPath of allCommandsPaths) {
+	const commandDir = path.basename(path.dirname(commandPath));
+	const heading = commandHeadings[commandDir] || formatTitle(commandDir);
+
+	if (!categoriesMap[heading]) {
+		categoriesMap[heading] = { title: heading, files: [] };
+		capitanoDoc.categories.push(categoriesMap[heading]);
+	}
+
+	categoriesMap[heading].files.push(commandPath);
```
Comment (myarmolinsky) [inline automation/capitanodoc/capitanodoc.ts:113]: Shouldn't we pushed to `categoriesMap[heading].files` first and *then* pushed to `capitanoDoc.categories`?

File: automation/build-bin.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-import type { JsonVersions } from '../lib/commands/version';
+import type { JsonVersions } from '../lib/commands/help/version';
```
[the remark this answers is not in the document]
Comment (vipulgupta2048) [reply automation/build-bin.ts:18]: It was how it was before and hence the structure. The command when they were root commands were documented together as `help & version`

I understand why it wouldn't work now and fixing it

File: automation/capitanodoc/capitanodoc.ts (hunk 2 above)
Comment (vipulgupta2048) [reply automation/capitanodoc/capitanodoc.ts:126]: Removing

File: automation/capitanodoc/capitanodoc.ts (hunk 5 above)
Comment (vipulgupta2048) [reply automation/capitanodoc/capitanodoc.ts:113]: You need to push an object first to `capitanoDoc.categories` having the `{title: "something", files: [ ] }`
I also was confused how exactly the `files` keeps updated on every loop instance but it seems that Node on Line 110 is really pushing a variable to `capitanoDoc.categories` than the actual value. I tried this here. 

```
➜  ~ node
> b = {a: [1]}
{ a: [ 1 ] }
> mega = {mega: b}
{ mega: { a: [ 1 ] } }
> b['a'].push(123)
2
> console.log(mega)
{ mega: { a: [ 1, 123 ] } }
```

Let me know if I have clarified incorrectly. This was a TIL for me too why this works but it does.

**Review by myarmolinsky — APPROVED** [review]

**Pull request merged**